// sam-resume/app/data.eric.js
/* eslint max-len: 'off' */

// Shorthand for line length
let skill = (name, level, time, icon) => ({
  name, level, time, class: icon
});

module.exports = {
  name:       'Eric Fonenault',
  paragraphs: [
    `Eric has a passion for mobile development and has been making people's
    lives better for the last 10 years ranging from network solutions for
    banks, to media libraries for online content delivery companies.`,
    `He is very conformant to the accepted practices of the language and team
    on which he works. His experience writing applications for Android and
    iOS (as well as Roku, Apple TV, and Chromecast) make him an invaluable
    member of the team.`,
    'Some of Eric\'s main strengths are:',
  ],
  favorites: {
    Agile:
      `Working in an agile environment at RightNow Media has brought
      productivity and teamwork to a new high. When Eric finishes his work, he
      switches languages and technologies to jump on board with the other
      members of the team and complete the current story on time.`,
    Deployment:
      `Extensive work with Team City, Jenkins, and Octopus Deploy has given
      Eric the foundational knowledge necessary to implement soft code updates
      over a vast network of devices.`,
    REST:
      `Working in online content delivery and the GameView servers has given
      Eric an appreciation of the REST protocol and best practices.`,
    Messaging:
      `Eric has previously worked with applications that needed to support real
      time messages in the social world. Whether sockets or long-polling, he
      understands the necessity of lightweight rapid message synchronization.`,
  },
  skills: [
    skill('Rabbit MQ',      '', '', 'rabbitmq'),
    skill('Octopus Deploy', '', '', 'octopusdeploy'),
    skill('TeamCity',       '', '', 'teamcity'),
    skill('Jenkins',        '', '', 'jenkins'),
    skill('Go',             '', '', 'golang'),
    skill('Webpack',        '', '', 'webpack'),
  ],
  languages: [
    skill('C#',         '', '', 'csharp'),
    skill('C++',        '', '', 'cplus'),
    skill('JavaScript', '', '', 'javascript'),
    skill('AngularJS',  '', '', 'angular'),
    skill('jQuery',     '', '', 'jquery'),
    skill('Java',       '', '', 'java'),
    skill('PHP',        '', '', 'php'),
    skill('SQL',        '', '', 'sql'),
    skill('HTML',       '', '', 'html'),
    skill('bash',       '', '', 'bash'),
    skill('PowerShell', '', '', 'powershell'),

  ],
  jobs: [
    {
      company: 'RightNow Media',
      time:    ['2015', 'Present'],
      title:   'Software Developer',
      skills:  [],
      things:  [
        `Utilized REST API to communicate between Application Interface server,
        Android, iPhone, Apple TV, Roku, and Chromecast devices.`,
        `Implemented several modules for the C# MVC backend in order to bring
        new products and features to an existing infrastructure. This brough
        the current technology current by incorporating some of the current
        best practices.`,
        `Developed scrollview, and grid scrollview for iOS to display product
        listings from the RightNow Media Application Interface server in a
        convenient familiar interface to native Apple users. The views utilized
        Apple transitions, which demonstrates a high level of experience.`,
      ],
    },
    {
      company: 'Heartland Payment Systems',
      time:    ['2015', '2013'],
      title:   'Programmer Analyst',
      skills:  [],
      things:  [
        `Successfully led a small team in rebuilding our Dispute Management
        website for the purposes of removing Silverlight and improving
        performance. We used an HTML/AngularJS frontend supported by a RESTful
        API built with ASP.NET’s Web Api 2 framework and were able to improve
        load times by optimizing code and significantly reducing the number of
        database requests.`,
        `Performed development on credit card transaction processing services
        that utilize an event-driven architecture backed by a RabbitMQ message
        broker. These systems were responsible for processing all of the
        transactions processed by Heartland Payment Systems.`,
        `Created an event-driven web app using javascript and the WebApi
        framework that captures errors in our transaction processing systems,
        providing Operations with an interface to identify major problems in the
        pipeline and reprocess failed transactions.`,
        `Worked with another developer to maintain our Silverlight-based web
        app, including bringing the app up to the latest PCI standards. This app
        is responsible for handling disputed transactions for all of our
        merchants.`,
        `Assisted a brand new junior developer by sharing my knowledge of
        programming whenever possible in a collaborative environment.`,
      ],
    },
    {
      company: 'VoidRay, LLC',
      time:    ['2013', '2013'],
      title:   'Software Developer',
      skills:  [],
      things:  [
        `Diligently applied myself to learning PHP and the Jackal PHP
        framework.`,
        `Collaborated with team members to create a PHP server that provides an
        API to retrieve pivot-table data sets from a customer’s product database.
        Using data access and abstraction layers, the API is able to provide a
        uniform interface while accounting for differences between product
        versions.`,
        `Using PHP, I assisted in developing a database forensics tools to aid in
        locating specific information and understanding how data is related.`,
        'Implemented bug fixes in customers’ custom C# applications.',
      ],
    },
    {
      company: 'Symfono Corporation',
      time:    ['2010', '2012'],
      title:   'Senior Software Developer',
      skills:  ['C++', 'PHP', 'BaSHellScript', 'TCL', 'Perl', 'AJAX', 'CSS', 'Objective-C', 'HTML', 'TCL', 'AWK', 'Java', 'GlassFish'],
      things:  [
        'Developed augmented reality application to enable people to see their Social Media friends overlaid on the real environment. The application also allowed the vendor to place objects, scripts, or advertisements in the environment.',
        'Developed server synchronization system to mirror data from arbitrary server into our own server.',
        'Wrote iOS GUI prototype that allowed development on the iPhone do download and run a GUI. This enabled faster development of the final application interface.',
        'Implemented streaming data compression algorithm to enable us to transmit binary data faster in a statically typed environment. This enabled us to display live data in a C++ application in a changing environment.',
        'Developed application server to transmit the scenegraph data to the iOS devices. The server utilized user roles to ensure appropriate access to the streamed data. The server also utilized versioning to ensure that the correct protocol was used for each device.',
      ],
    },
    {
      company: 'Symfono',
      time:    ['2012', '2013'],
      title:   'Senior Software Developer',
      skills:  [],
      things:  [
        `Demonstrated a passionate level of commitment and a willingness to
        further develop skills in several areas while working as a member of a
        team responsible for the creation of the following products:`,
        `Using PureMVC to shape our C, C++, and Objective-C code, my team
        created an iPad app that visualizes match data for professional and
        international soccer games. This innovative technology provides a new
        and immersive way for fans to follow their favorite teams. This app is
        currently in the iTunes App Store.`,
        `The GameView Engine is the web server that drives the GameView app.
        Using Java, MySQL, and GlassFish, we transform and interpret the data
        stream containing soccer game data from our 3rd party statistics
        provider and encode in the Hessian Binary Web Service Protocol to be
        transmitted to the client.`,
        `Also driven by the GameView Engine, our GameView API allows clients to
        embed a JavaScript port of our GameView product into their websites. `,
      ],
    },
    {
      company: 'InTouch Credit Union',
      time:    ['2009', '2012'],
      title:   'Network Technician',
      skills:  [],
      things:  [
        `Designed and built a database-driven system using SQL Sever 2010
        Express and Windows PowerShell for keeping track of all active
        workstations on our network, tracking the progress of workstation-level
        projects, as well as recording and managing the security IDs for our
        core financial system.`,
        `Created an "IT Dashboard" as a centralized tool for computer and AD
        account maintenance using the .NET features in Windows PowerShell.`,
        `Streamlined several time-consuming manual processes through use of
        Windows PowerShell, such as renaming printers on remote workstations,
        managing the Proxy Server setting in IE at the branch level, and
        scripting the installation and customization of our new core financial
        system.`,
        `Successfully tasked with maintaining the health of our physical network
        at all 20 branch locations.`,
        `Managed Active Directory, created Group Policy Objects, and maintained
        our DNS server on Server 2003 and Server 2008 R2.`,
        `Demonstrated adept troubleshooting skills in diagnosing and repairing
        Windows XP and Windows 7 computers.`,
        `Active member of an internal committee designed to analyze current
        processes to cut costs, drive revenue, and increase employee
        retention.`,
      ],
    },
    {
      company: 'Ameripath',
      time:    ['2008', '2009'],
      title:   'Level 1 and 2 Support Technician',
      skills:  ['HTML', 'Java', 'Javascript', 'C++', 'Cobal', 'Oracle', 'VBScript', 'JScript', 'MS Access', 'WMI', 'Lisp', 'AutoCad', 'Visio', 'MicroStation'],
      things:  [
        `Maintained the IT Helpdesk by providing accurate and insightful support
        for Windows XP and Office 2003, giving phone and email coverage,
        creating and following up on problem tickets, troubleshooting VPN and
        printer issues, and documenting the flow of information between specific
        service desks and the affected users. `,
        `Frequently utilized Remote Desktop and Dameware to access remote
        systems to effectively diagnose and resolve OS and software issues. `,
        `Responsible for the creation and maintenance of Active Directory and
        Exchange accounts.`,
        `Demonstrated ability to listen to clients’ issues and identify the root
        problem.`,
        `Displayed patience and respect while working through stressful
        situations with clients`,
        `Showed the ability to learn quickly while working to understand company
        protocols and extensive system structure`,
        `Employed excellent problem solving skills through diagnosing and
        repairing hardware and software issues`,
      ],
    },
  ],
  education: [{
    name:    'Grove City College, Grove City, PA',
    degrees: [
      {name: 'B.S. Computer Science', year: 2007},
      {name: 'B.S. Computer Information Systems', year: 2007},
    ]
  }],
};
