// sam-resume/app/data.sam.js
/* eslint max-len: 'off' */

// Shorthand for line length
let skill = (name, level, time, icon) => ({
  name, level, time, class: icon
});

module.exports = {
  name:       'Samuel Dillow',
  paragraphs: [
    `The main thing people ask me is what language do you program in. Strange
    question, as most good programmers can program in nearly any language. I
    started programming when I was 8, so I have racked up experience in over
    40 languages to this point.`,
    `In the United States Air Force I learned about leadership and the
    importance of structure. Combining my passion for programming and the
    experience of the US Military I have led several teams in the development
    and delivery of disparate software solutions.`,
    'However, my favorite languages are C++, NodeJS, and Java.',
  ],
  favorites: {
    'C++':
      `I use C++ when I write command line utilities that need to do things
      that regular scripts cannot. I write about 6 command line programs per
      year.`,
    'NodeJS':
      `Since V8 has revamped the JavaScript world, I've been writing
      applications in JavaScript. I find it to be tons faster and more stable.
      Now that NodeJS has a CLR, it's an excellent platform to write fast and
      responsive cross-platform code. I've written server apps with it, and
      used Electron (and nw.js) to write GUI apps.`,
    'Java':
      `Java has a lot of trouble doing operating system specific tasks, so
      it's great for writing a cross-platform application. I use Java for
      command line utilities and web servers. I write about 1 Java program per
      month.`,
    'TCL':
      `I use TCL / Expect to write anything that has to be done with CLI, but
      takes up more than one line. Bash / Zsh scripts usually can do the
      trick, but many times TCL ends up creating a much more intelligent
      product. I write about 1 TCL script per week.`
  },
  skills: [
    skill('Meteor',      'Advanced', '', 'meteor'),
    skill('Apache HTTP', 'Expert',   '', 'apache-http'),
    skill('Glassfish',   'Advanced', '', 'glassfish'),
    skill('NodeJS',      'Expert',   '', 'nodejs'),
    skill('Linux',       'Advanced', '', 'linux'),
    skill('MS Access',   'Advanced', '', 'ms-access'),
    skill('MySQL',       'Advanced', '', 'mysql'),
    skill('Oracle',      'Good',     '', 'oracle'),
    skill('PureMVC',     'Advanced', '', 'puremvc'),
    skill('SQL Server',  'Advanced', '', 'sql-server'),
    skill('Ansible',     'Advanced', '', 'ansible'),
    skill('Java Swing',  'Advanced', '', 'swing'),
    skill('iOS',         'Advanced', '', 'ios'),
  ],
  languages: [
    skill('NodeJS',         'Expert',   '', 'nodejs'),
    skill('Angular',        'Expert',   '', 'angular'),
    skill('Sass',           'Advanced', '', 'sass'),
    skill('Less CSS',       'Advanced', '', 'less'),
    skill('jQuery',         'Expert',   '', 'jquery'),
    skill('JavsScript',     'Expert',   '', 'javascript'),
    skill('AJAX/DHTML',     'Expert',   '', 'ajax-dhtml'),
    skill('ASP',            'Expert',   '', 'asp'),
    skill('ASP.NET',        'Advanced', '', 'asp-net'),
    skill('AWK',            'Good',     '', 'awk'),
    skill('ActionScript 2', 'Expert',   '', 'actionscript-2'),
    skill('ActionScript 3', 'Expert',   '', 'actionscript-3'),
    skill('AutoHotkey',     'Expert',   '', 'autohotkey'),
    skill('BaSHellScript',  'Advanced', '', 'bash'),
    skill('C / C++',        'Advanced', '', 'cplus'),
    skill('C Sharp (C#)',   'Advanced', '', 'csharp'),
    skill('CSS',            'Expert',   '', 'css'),
    skill('Cold Fusion',    'Good',     '', 'cold-fusion'),
    skill('Flex',           'Expert',   '', 'flex'),
    skill('GWBasic',        'Expert',   '', 'gwbasic'),
    skill('HTML 4',         'Expert',   '', 'html'),
    skill('HTML 5',         'Expert',   '', 'html5'),
    skill('JScript',        'Advanced', '', 'jscript'),
    skill('Java',           'Good',     '', 'java'),
    skill('LISP',           'Good',     '', 'lisp'),
    skill('Objective-C',    'Fair',     '', 'objective-c'),
    skill('PHP',            'Expert',   '', 'php'),
    skill('Pascal',         'Good',     '', 'pascal'),
    skill('Perl',           'Good',     '', 'perl'),
    skill('QBasic',         'Expert',   '', 'qbasic'),
    skill('PowerShell',     'Fair',     '', 'powershell'),
    skill('RequireJS',      'Expert',   '', 'requirejs'),
    skill('SGML',           'Expert',   '', 'sgml'),
    skill('TCL',            'Advanced', '', 'tcl'),
    skill('VBScript',       'Expert',   '', 'vbscript'),
    skill('Visual Basic',   'Advanced', '', 'visual-basic'),
    skill('XML',            'Expert',   '', 'xml'),
    skill('haXe',           'Advanced', '', 'haxe'),
  ],
  jobs: [
    {
      company: 'VoidRay Co',
      time:    ['2012', '2016'],
      title:   'Senior Software Developer',
      skills:  ['C++', 'PHP', 'BaSHellScript', 'Ansible', 'TCL', 'Perl', 'AJAX', 'CSS', 'Objective-C', 'HTML', 'TCL', 'AWK', 'Java', 'GlassFish', 'Angular'],
      things:  [
        'Engineered NodeJS MySQL WebSockets system to allow secure streaming of SQL data through a firewall. This allowed us to present custom reports to clients who had no way to provide access to their internal database.',
        'Developed system to translate Google Analytics into graphs that real people can use. The server is written in NodeJS using JSON Web Tokens (JWT) to enable rapid scaling. It communicates with Google Analytics via oAuth. The web interface is written in Angular, using D3 for the responsive graphs.',
        'Wrote system to migrate old inventory system from SQL to Mongo. As a result, we managed to migrate a customer E-Commerce application into Shopify one third of the time of the competition.',
        'Wrote system to scrape financial data from a bank for private use. The bank api was severely lacking in manner of presentation and interaction. This app saved the customer over $78,000 per year.',
      ],
    },
    {
      company: 'KloudNation',
      time:    ['2015', '2015'],
      title:   'Software Developer',
      skills:  ['C++', 'NodeJS', 'Ansible', 'Java', 'SaltStack', 'AJAX', 'CSS', 'Objective-C', 'Swift', 'HTML', 'Angular', 'Meteor'],
      things:  [
        'Created job processing network that auto-scaled it\'s topology between 20 and 200 servers based on the demand.',
        'Developed website for merging and comparing datasets from different vendors and generating reports based on the results.',
        'Developed augmented reality app for iPhone and Android that used beacons to perform translateration. This resulted in precision tracking of customers at an event while providing an immersive experience at the same time.',
      ],
    },
    {
      company: 'Symfono Corporation',
      time:    ['2010', '2012'],
      title:   'Senior Software Developer',
      skills:  ['C++', 'PHP', 'BaSHellScript', 'TCL', 'Perl', 'AJAX', 'CSS', 'Objective-C', 'HTML', 'TCL', 'AWK', 'Java', 'GlassFish'],
      things:  [
        'Developed augmented reality application to enable people to see their Social Media friends overlaid on the real environment. The application also allowed the vendor to place objects, scripts, or advertisements in the environment.',
        'Developed server synchronization system to mirror data from arbitrary server into our own server.',
        'Wrote iOS GUI prototype that allowed development on the iPhone do download and run a GUI. This enabled faster development of the final application interface.',
        'Implemented streaming data compression algorithm to enable us to transmit binary data faster in a statically typed environment. This enabled us to display live data in a C++ application in a changing environment.',
        'Developed application server to transmit the scenegraph data to the iOS devices. The server utilized user roles to ensure appropriate access to the streamed data. The server also utilized versioning to ensure that the correct protocol was used for each device.',
      ],
    },
    {
      company: 'Odyssey Communications Group',
      time:    ['2009', '2010'],
      title:   'Senior Software Developer',
      skills:  ['C++', 'PHP', 'BaSHellScript', 'TCL', 'Perl', 'AJAX', 'CSS', 'Flex', 'Objective-C', 'ActionScript 3', 'HTML', 'TCL', 'AWK', 'Cold Fusion'],
      things:  [
        'Architected a digital scoreboard server and client system to track real-time game information across a network. My knowledge of PureMVC and ActionScript 3 were critical in the successful completion and stability of this project.',
        'Routed out a significant long-standing flaw which was crippling server stability. My decisive resolution prevented the loss of critical data, and saved 8 hours of lost time per week demonstrating highly capable problem solving skills as well as a keen knowledge of OSX core structure.',
        'Played a critical role in the smooth migration of 600 customers to new server with a total of less than 7 minutes downtime. The new server provided more scalability and speed as well as preventing the eminent corruption of customer data.',
      ],
    },
    {
      company: 'Pursuant Group',
      time:    ['2008', '2009'],
      title:   'Software Developer',
      skills:  ['AJAX', 'XML', 'Flex', 'ASP.net', 'ActionScript 3', 'HTML', 'VBScript'],
      things:  [
        'Worked as part of a team of 15 highly experienced developers to produce a rich internet application for social networking to the the flagship product for the company. Managed to bring team of different backgrounds together with emphasis on fostering teamwork and effective communication.',
        'Attended class on Flex and PureMVC and brought the information back to present to the team. This resulted in a more skilled and cohesive workforce and saved $6,000 in training costs.',
      ],
    },
    {
      company: 'United States Air Force',
      time:    ['2003', '2008'],
      title:   'Communications and Computer Systems Project Manager',
      skills:  ['HTML', 'Java', 'JavaScript', 'C++', 'Cobal', 'Oracle', 'VBScript', 'JScript', 'MS Access', 'WMI', 'Lisp', 'AutoCad', 'Visio', 'MicroStation'],
      things:  [
        'Trained 600 persons in use of Oracle based telecommunications management software',
        'Upgraded Telecommunications Management Server to interface with more than 27,000 phones and computer systems across base network.',
        'Implemented IPv6 compatible network, running voice and data over same infrastructure in preparation for Voice Over IP technology.',
        'Managed communications infrastructure records showing overall network topology and 3D geospatial network structure including internal facility wiring.',
        'Executed 7 projects for installing aircraft monitoring systems in conjunction with base network.',
        'Established new airfield security system, rivaling 3 previously failed attempts by security team.',
        'Taught seven computer programming classes in the object oriented languages C++ and Java.',
        'Taught classes for combat preparation, weapons utilization and maintenance, and tactics.',
        'Certified CPR Instructor',
        'Instructed classes on Field Triage, First Aid, Biological Defense, and Explosive Ordinance',
      ],
    },
    {
      company: 'Cantex Management',
      time:    ['2001', '2003'],
      title:   'Information Technology',
      skills:  ['SGML', 'HTML', 'XML', 'Flash', 'Assembly Language', 'Java', 'VBScript', 'JScript', 'MySQL'],
      things:  [
        'Managed Microsoft Windows Server 2003 with less than 50 clients. Created and managed net-centric video surveillance system to monitor retail stores in 7 states. Developed electronic interactive training system for employees with home-accessible web based interface',
      ],
    },
    {
      company: 'Nortel Networks',
      time:    ['1999', '2001'],
      title:   'Training and Documentation Department',
      skills:  ['AJAX', 'XML', 'Flex', 'ASP.net', 'ActionScript 3', 'HTML', 'VBScript'],
      things:  [
        'Provided classes on authoring by use of SGML technology Migrated 500 users from Microsoft Works to Adobe FrameMaker SGML Created software to convert between Microsoft Works, SGML, HTML, XML, and PostScript. Collaborated with Tiger Team for establishing web-based training site Created web-based Trouble Management System to integrate with Remedy Software and decentralize help-desk operations.',
      ],
    },
  ],
};
