// sam-resume/app/data.chris.js
/* eslint max-len: 'off' */

// Shorthand for line length
let skill = (name, level, time, icon) => ({
  name, level, time, class: icon
});

module.exports = {
  name:       'Chris Byler',
  paragraphs: [
    `His degree in Electrical Engineering will be critical for establishing a
    stable charging environment and dealing with communication between the
    Charge Point and the actual charging hardware.`,
    `For fun, he developed a backup sensor for a real-world vehicle during his
    time at the University of Texas at Dallas. The motion sensor was written for
    the Texas Instruments MSP 430 and utilized an onboard wireless antenna to
    communicate with the rest of the instrumentation of the vehicle.`,
    `Chris was in the Math club of the University of Texas at Dallas. His
    ability to crunch numbers in his head while working is extremely helpful
    when working in teams.`,
  ],
  favorites: {
    'Team Player':
      `Currently Chris works in a team of 30 software developers and integrates
      with them on a daily basis to help the entire team complete tasks on time.
      Previous work as a Production Support Manager heightens his strength as
      a team member in understanding that the health of the whole team is what
      ultimately results in a successful deployment.`,
    'User Interface':
      `Chris has a passion for developing useful interfaces for his users. He
      relates strongly to the frustrations of users pawing ineffectually at the
      screens presented by primitive systems. He has created control panels in
      AS3, React, OpenGL, C++, and Swing`,
    'Wireless':
      `His work in electrical engineering has given him experience working with
      low-level antennae and bluetooth networking, which will be ciritcal for
      ensuring quality development of the connectivity feature of the in-home
      charger.`,
  },
  skills: [
    skill('Android',                     '', '', 'android'),
    skill('Away3D',                      '', '', 'away3d'),
    skill('Bluetooth',                   '', '', 'bluetooth'),
    skill('Drupal',                      '', '', 'drupal'),
    skill('Enterprise Java Beans (EJB)', '', '', 'ejb'),
    skill('FLUX',                        '', '', 'flux'),
    skill('Glassfish',                   '', '', 'glassfish'),
    skill('iOS',                         '', '', 'ios'),
    skill('MVC',                         '', '', 'mvc'),
    skill('MVP',                         '', '', 'mvp'),
    skill('OpenGL',                      '', '', 'opengl'),
    skill('RequireJS',                   '', '', 'requirejs'),
    skill('Underscore',                  '', '', 'underscore'),
  ],
  languages: [
    skill('ActionScript 3 (AS3)',        '', '', 'actionscript-3'),
    skill('HTML',                        '', '', 'html'),
    skill('HTML5',                       '', '', 'html5'),
    skill('JavaScript',                  '', '', 'javascript'),
    skill('C#',                          '', '', 'csharp'),
    skill('PHP',                         '', '', 'php'),
    skill('React',                       '', '', 'react'),
    skill('Java',                        '', '', 'java'),
    skill('ObjectiveC',                  '', '', 'objective-c'),
    skill('Swing',                       '', '', 'swing'),
    skill('C++',                         '', '', 'cplus'),
  ],
  jobs: [
    {
      company: 'One Network Enterprises',
      time:    ['Present', '2013'],
      title:   'Software Engineer',
      skills:  [],
      things:  [
        `Software Engineer ­ Developer on the Platform team from which the core
        technology stemmed in the company.`,
        `New Front End UI ­ Developed and integrated visual enhancements for the
        main UI of the technology. Worked with the in house designer to
        implement the desired look and feel of the application while providing
        suggestions for further enhancements.`,
        `3D Framework ­ Scoped and worked on implementing a 3D framework to
        model the company’s data model for representing a Supply Chain. This
        included writing a working system for Flash + Away3D at first, then
        reformatting to the suggested platform of Unity. This interface includes
        2D and 3D elements with interactive components that a user can
        manipulate to seamlessly interact with the backend data. This seamless
        interaction worked via a complicated bridging system between plugin and
        browser javascript which communicated to a Java server backend for
        persistence.`,
        `Design Patterns (New Framework RnD) ­ Encouraged and worked with
        co­workers to develop a new methodology of implementing new UIs by
        introducing design patterns to the legacy system. This has led to the
        exploration and implementation of new features in an MVP and FLUX
        pattern while testing and trying modern frameworks such as React,
        Requirejs, Underscore, and many others to find the ideal pipeline. We
        used our research to integrate older legacy implementations to the newer
        patterns.`,
        `Build Processes and new Javascript Standards ­ Engineered and
        integrated build processes to include a workflow for utilizing new Web
        standards such as ES6 and module support. This had many technical
        requirements to work with legacy systems and support a crossover of CMD
        and AMD module loading to allow for partial bundling of the Client’s SPA
        code base. This also included integrating ReactJS with the system and
        support injecting other third party libraries into the company’s
        workflow.`,
      ],
    },
    {
      company: 'Symfono, Inc',
      time:    ['2012', '2013'],
      title:   'Software Engineer',
      skills:  [],
      things:  [
        `iOS Soccer App ­ Implemented code for creating a mobile app that
        communicated with a data server to present soccer games play by play in
        an easy to understand interface which required over 70,000 lines of
        stable code in Objective­C, C++, JavaScript, HTML, and Java that adhered
        to a MVC code pattern.`,
        `Java Server ­ Implemented a java Glassfish server that hosted data to a
        large scale client base which incorporated EJB and JPA code patterns.`,
        `Data Standard Parsing ­ Created a STAX parser that analyzed a
        complicated data structure from Opta that detailed out soccer games and
        saved the data in a format that could be served to clients to view
        graphical representations of games being played.`,
        `Cross Platform ­ Implemented a code design that bridged Objective­c and
        C++ where the project could be broken into cross platform and OS
        dependent pieces to improve portability.`,
        'OpenGL API ­ Developed an API in objective­c for working with OpenGL.',
        `Augmented Reality ­ Researched and implemented use of monocular (single
        camera) technologies for mapping real life environments for use in
        Augmented Reality applications.`,
      ],
    },
    {
      company: 'Caltrop Interactive',
      time:    ['2011', '2012'],
      title:   'Software Engineer',
      skills:  [],
      things:  [
        `Web Development ­ Developed website for potential Political Candidate
        using Drupal Framework.`,
        `Research ­ Fulfilled a contract with Raytheon to draw up a research
        document whose nature is under NDA with the company.`,
        `Android App ­ Fulfilled a contract with Raytheon to develop an Android
        mobile application to aid in streamlining processes. Details on this
        project are under NDA.`,
        `Bluetooth Network ­ Worked with Bluetooth networking using the Android
        development kit where thorough knowledge of Bluetooth specifications was
        used to implement networking capabilities between numerous devices.`,
      ],
    },
    {
      company: 'Consulting',
      time:    [],
      title:   'Software Engineer',
      skills:  [],
      things:  [
        `Professional Code Review ­ Performed code reviewing for the GeoTag
        patent court case for patent number 5,930,474 which was brought against
        hundreds of defendants.`,
        `Web Development ­ Worked on numerous websites, helping with
        implementing the site using php, java, C#, javascript, and Flash to make
        the required designs a reality.`,
        `American Airlines ­ Developed the flash components needed for a
        MetroPCS “What Phone Are You” app that was used in a kiosk display in
        the American Airlines Center.`,
        `Sour Jacks ­ Developed the iPad “pucker up” application for a company
        that needed a specific graphical distortion to photos taken with the
        ipad which were then sent to a server which printed the images for the
        company’s clients at a kiosk booth where they were advertising.`,
      ],
    },
  ],
  education: [{
    name:    'University of Texas at Dallas',
    degrees: [
      {name: 'B.S. in Electrical Engineering', year: 2010},
    ]
  }],
};
